config/backend-config/multitenant_workspaces.go

```diff
@@ -74,6 +74,11 @@ func (workspaceConfig *MultiTenantWorkspacesConfig) Get(workspaces string) (Conf
 
 // getFromApi gets the workspace config from api
 func (workspaceConfig *MultiTenantWorkspacesConfig) getFromAPI(workspaceArr string) (ConfigT, bool) {
+	// added this to avoid unnecessary calls to backend config and log better until workspace IDs are not present
+	if workspaceArr == workspaceConfig.Token {
```

Comment:
maybe we need to init with an empty string instead of `workspaceConfig.Token`.
This check here takes some time to understand and would be easy to break with a refactor

Reply:
done

Reply:
This will never be equal right ? Since Token is essentially `HOSTED_MULTITENANT_SERVICE_SECRET` ?

Reply:
right, so the first time when `startWithIds` gets called from main then we call it with the token (due to 3 type of backend config possibilities).

Reply:
So the setup call from main would be having this right? and if it's the same then we would skip